otter/integration/tests/test_pause.py

```diff
@@ -0,0 +1,159 @@
+"""
+Tests covering `../groups/groupId/pause` and `../groups/groupId/resume`
+endpoints
+"""
+from testtools.matchers import ContainsDict, Equals
+
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
+from twisted.trial import unittest
+
+from otter.integration.lib.autoscale import ScalingPolicy
+from otter.integration.lib.mimic import MimicNova
+from otter.integration.lib.resources import TestResources
+from otter.integration.lib.trial_tools import (
+    TestHelper,
+    get_identity,
+    get_resource_mapping,
+    get_utcstr_from_now,
+    otter_build_timeout,
+    region,
+    scheduler_interval,
+    sleep
+)
+
+
+timeout_default = 600
+
+
+class PauseTests(unittest.TestCase):
+    """
+    Tests for `../groups/groupId/pause` endpoint
+    """
+
+    skip = "Until #1604 is implemented"
+
+    def setUp(self):
+        self.helper = TestHelper(self)
+        self.rcs = TestResources()
+        self.identity = get_identity(self.helper.pool)
+        return self.identity.authenticate_user(
+            self.rcs,
+            resources=get_resource_mapping(),
+            region=region
+        )
+
+    @inlineCallbacks
+    def test_pause_getstate(self):
+        """
+        A paused group's state will say paused:True
+        """
+        group, _ = self.helper.create_group()
+        yield group.start(self.rcs, self)
+        yield self.helper.assert_group_state(
+            group, ContainsDict({"paused": Equals(False)}))
+        yield group.pause(self.rcs)
+        yield self.helper.assert_group_state(
+            group, ContainsDict({"paused": Equals(True)}))
+
+    @inlineCallbacks
+    def test_pause_stops_convergence(self):
+        """
+        Pausing a group will stop any further convergence cycle. We do this by
+        1. Setup mimic to build server for just before timeout
+        2. Creating a group with 1 min entity
+        3. Pause the group
+        4. Wait for server to finish building in mimic
```

Comment:
Should this test only be run in the mimic environment?  It seems like with a few changes, it could also work in production.

1.  If it's mimic only, this test should probably have a `@skip_if` decorator.
2.  If we want this to also work in production, maybe we can poll Nova to wait until Nova says the server is active, then wait some more time for a convergence cycle to kick off, and then assert something about the state?

Possibly even on mimic though this could spuriously pass, if pause does not stop convergence cycles:  If the server on mimic goes active at time `t=otter_build_timeout - 5`, and a convergence cycle happened right before that, then if we check the state at time `t=otter_build_timeout`, another convergence cycle may not have happened yet (but will) and hence it'd still say that the serve was building.  

To be sure that convergence cycles have stopped, should we wait at least 1 convergence cycle of time?